### PR TITLE
feat(notifications): added Mark All Read bulk action

### PR DESCRIPTION
**Pull request by mareklibra**
The user can newly mark all unread messages as read at one click.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by mareklibra** [comment]
[Screencast from 2024-04-23 10-35-10.webm](https://github.com/backstage/backstage/assets/17194943/a35f9395-d0c8-45f7-9a0d-f94f4632bbd7)

File: plugins/notifications/src/components/NotificationsTable/NotificationsTable.tsx [hunk 1]
```diff
@@ -117,6 +123,31 @@ export const NotificationsTable = ({
...
+        ).notifications?.map(notification => notification.id);
+
+        return notificationsApi
+          .updateNotifications({
+            ids,
+            read: true,
+          })
+          .then(onUpdate);
+      })
+      .catch(noop);
```
Comment (drodil) [inline plugins/notifications/src/components/NotificationsTable/NotificationsTable.tsx:148]: Maybe instead noop, use the alertApi to show user an error?

**Comment by drodil** [comment]
Looks good, just minor thing about handling errors

File: plugins/notifications/src/components/NotificationsTable/NotificationsTable.tsx (hunk 1 above)
Comment (mareklibra) [reply plugins/notifications/src/components/NotificationsTable/NotificationsTable.tsx]: done

**Review by freben — APPROVED** [review]

**Pull request merged**